lib/spack/spack/spack_deps.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import contextlib
+import os
+import sys
+
+import llnl.util.filesystem as fs
+import llnl.util.tty as tty
+
+import spack.spec
+import spack.store
+import spack.user_environment as uenv
+import spack.util.executable
+from spack.util.environment import EnvironmentModifications
+
+
+@contextlib.contextmanager
+def system_python_context():
+    python_cls = type(spack.spec.Spec('python').package)
+    python_prefix = os.path.dirname(os.path.dirname(sys.executable))
+    externals = python_cls.determine_spec_details(
+        python_prefix, [os.path.basename(sys.executable)])
+    external_python = externals[0]
+
+    entry = {
+        'buildable': False,
+        'externals': [
+            {'prefix': python_prefix, 'spec': str(external_python)}
+        ]
+    }
+
+    with spack.config.override('packages:python::', entry):
+        yield
+
+
+def make_module_available(module, spec=None, install=False):
+    """Ensure module is importable"""
+    # If we already can import it, that's great
+    try:
+        __import__(module)
+        return
+    except ImportError:
+        pass
+
+    # If it's already installed, use it
+    # Search by spec
+    spec = spack.spec.Spec(spec or module)
+
+    # We have to run as part of this python
+    spec.constrain('^python@%d.%d' % sys.version_info[:2])
```

Comment:
Does this syntax actually work? I know we've had a lot of problems with `X.Y` vs `X.Y.Z` in the past. That's why we've always had to do `python@2.7.0:2.7.999`

Reply:
Good point. It works for queries, but not for building because it reads as concrete.

Reply:
Wait no this should be fine -- it's used as a query, where it doesn't cause a problem, and it's used in a context manager in which the current python is an external and python isn't buildable. So there isn't the problem of concretizing to a partial version. I'll add a comment to the code to that effect though, since the reason it works is a little subtle.

Reply:
Just relatedly, I had tried to introduce a new version syntax e.g. `2.6.*` to give people something familiar , then immediately  removed it upon user feedback about how it conflicts with shell metacharacters: https://github.com/spack/spack/pull/20258#issuecomment-739553142

However, note that that PR also introduces 3 new version range operators `!`, `!:`, and `:!` to the spec syntax, which imo would provide a familiar-looking encoding of this specific common type of constraint, as **`2.6:!2.7`**. We could even consider raising a (silencable) DeprecationError whenever we parse any version string containing e.g. `.999` or `.0.0.1`, and print a command line to fix it?

Reply:
I assume yaml is easier to mechanically transform than python itself, making it an easier prospect than e.g. the equivalent in pants -- see pantsbuild/pants#9434 for a PR i still havent merged yet.

Reply:
But also of note, libCST (scripts? util libraries?) would be a great way to implement both automated upgrades to package.py files (if ever needed), as well as "shading" import paths to expose multiple versions of a python package within the same running interpreter.